src/Files.App/ViewModels/UserControls/StatusCenterViewModel.cs

```diff
@@ -3,148 +3,113 @@
 
-		public OngoingTasksViewModel()
+		public StatusCenterViewModel()
 		{
-			StatusBannersSource.CollectionChanged += (s, e) => OnPropertyChanged(nameof(AnyBannersPresent));
+			StatusCenterItems.CollectionChanged += (s, e) => OnPropertyChanged(nameof(HasAnyItem));
 		}
 
-		public PostedStatusBanner PostBanner(string title, string message, int initialProgress, ReturnResult status, FileOperationType operation)
+		public StatusCenterItem AddItem(string title, string message, int initialProgress, ReturnResult status, FileOperationType operation, CancellationTokenSource cancellationTokenSource = null)
 		{
-			StatusBanner banner = new StatusBanner(message, title, initialProgress, status, operation);
-			PostedStatusBanner postedBanner = new PostedStatusBanner(banner, this);
+			var banner = new StatusCenterItem(message, title, initialProgress, status, operation, cancellationTokenSource);
 
-			StatusBannersSource.Insert(0, banner);
-			ProgressBannerPosted?.Invoke(this, postedBanner);
+			StatusCenterItems.Insert(0, banner);
+			NewItemAdded?.Invoke(this, banner);
 
-			UpdateBanner(banner);
+			NotifyChanges();
 
-			return postedBanner;
+			return banner;
 		}
 
-		public PostedStatusBanner PostOperationBanner(string title, string message, int initialProgress, ReturnResult status, FileOperationType operation, CancellationTokenSource cancellationTokenSource)
+		public bool RemoveItem(StatusCenterItem banner)
 		{
-			StatusBanner banner = new(message, title, initialProgress, status, operation)
-			{
-				CancellationTokenSource = cancellationTokenSource,
-			};
-
-			PostedStatusBanner postedBanner = new(banner, this, cancellationTokenSource);
-
-			StatusBannersSource.Insert(0, banner);
-			ProgressBannerPosted?.Invoke(this, postedBanner);
-
-			UpdateBanner(banner);
-
-			return postedBanner;
-		}
-
-		public PostedStatusBanner PostActionBanner(string title, string message, string primaryButtonText, string cancelButtonText, Action primaryAction)
-		{
-			StatusBanner banner = new(message, title, primaryButtonText, cancelButtonText, primaryAction);
-			PostedStatusBanner postedBanner = new(banner, this);
-
-			StatusBannersSource.Insert(0, banner);
-			ProgressBannerPosted?.Invoke(this, postedBanner);
-
-			UpdateBanner(banner);
-
-			return postedBanner;
-		}
-
-		public bool CloseBanner(StatusBanner banner)
-		{
-			if (!StatusBannersSource.Contains(banner))
+			if (!StatusCenterItems.Contains(banner))
 				return false;
 
-			StatusBannersSource.Remove(banner);
+			StatusCenterItems.Remove(banner);
 
-			UpdateBanner(banner);
+			NotifyChanges();
 
 			return true;
 		}
 
-		public void CloseAllBanner()
+		public void RemoveAllCompletedItems()
 		{
-			for (int i = StatusBannersSource.Count - 1; i >= 0; i--)
+			StatusCenterItems.ForEach((x) =>
 			{
-				var itemToDismiss = StatusBannersSource[i];
-				if (!itemToDismiss.IsProgressing)
-					CloseBanner(itemToDismiss);
-			}
+				if (!x.IsInProgress)
+					RemoveItem(x);
+			});
```

Comment:
`StatusCenterItems.RemoveAll(x => !x.IsNoProgress)` to make it shorter if I understand well?

Reply:
ObservableCollection doesn't contain defnition of RemoveAll. So converted ToList first. done.